File: Polimorfismo/Ejercicio5/Python/Universidad.py
class Oficina:
    def __init__(self, nroSillas, nroEscritorios, nroEstanterias):
        self.nroSillas = nroSillas
        self.nroEscritorios = nroEscritorios
        self.nroEstanterias = nroEstanterias
    def cantidadDeMuebles(self):
        print("La cantidad de muebles es de: ", self.nroSillas + self.nroEscritorios + self.nroEstanterias)
    def mostrar(self):
        print("La cantida de sillas es: ", self.nroSillas, "la cantidad de escritorios es de", self.nroEscritorios, "la cantidad de estanterias es", self.nroEstanterias)
class Aula:
    def __init__(self, nombre, capacidad, nroPupitres):
        self.nombre = nombre
        self.capacidad = capacidad
        self.nroPupitres = nroPupitres
    def cantidadDeMuebles(self):
        print("La cantidad de muebles es de: ", self.nroPupitres)
    def mostrar(self):
        print("el aula se llama: ", self.nombre, "la capacidad es para", self.capacidad, "la cantidad de pupitres es", self.nroPupitres)
class Laboratorio:
    def __init__(self, nombre, capacidad, nroMesas, nroSillas):
        self.nombre = nombre
        self.capacidad = capacidad
        self.nroMesas = nroMesas
        self.nroSillas = nroSillas
    def cantidadDeMuebles(self):
        print("La cantidad de muebles es de: ",self.nroMesas + self.nroSillas)
    def mostrar(self):
        print("el laboratorio se llama: ", self.nombre, "la capacidad es de", self.capacidad, "la cantidad de mesas es", self.nroMesas, "y la cantidad de sillas es de", self.nroSillas)
oficina1 = Oficina(50, 25, 8)
oficina2 = Oficina(35, 65, 15)
aula1 = Aula("A3-210519", 150, 150)
aula2 = Aula("A1-210409", 100, 100)
laboratorio1 = Laboratorio("A1-Lab1", 150, 75, 150)
oficina1.mostrar()
oficina2.mostrar()
aula1.mostrar()
aula2.mostrar()
laboratorio1.mostrar()
oficina1.cantidadDeMuebles()
oficina2.cantidadDeMuebles()
aula1.cantidadDeMuebles()
aula2.cantidadDeMuebles()
laboratorio1.cantidadDeMuebles()

File: Polimorfismo/Ejercicio5/Java/Universidad.java
class Oficina {
    private int nroSillas;
    private int nroEscritorios;
    private int nroEstanterias;
    public Oficina(int nroSillas, int nroEscritorios, int nroEstanterias) {
        this.nroSillas = nroSillas;
        this.nroEscritorios = nroEscritorios;
        this.nroEstanterias = nroEstanterias;
    }
    public void cantidadDeMuebles() {
        System.out.println("La cantidad de muebles es de: " + (nroSillas + nroEscritorios + nroEstanterias));
    }
    public void mostrar() {
        System.out.println("La cantidad de sillas es: " + nroSillas + ", la cantidad de escritorios es de: " + nroEscritorios + ", la cantidad de estanterías es: " + nroEstanterias);
    }
}
class Aula {
    private String nombre;
    private int capacidad;
    private int nroPupitres;
    public Aula(String nombre, int capacidad, int nroPupitres) {
        this.nombre = nombre;
        this.capacidad = capacidad;
        this.nroPupitres = nroPupitres;
    }
    public void cantidadDeMuebles() {
        System.out.println("La cantidad de muebles es de: " + nroPupitres);
    }
    public void mostrar() {
        System.out.println("El aula se llama: " + nombre + ", la capacidad es para: " + capacidad +  ", la cantidad de pupitres es: " + nroPupitres);
    }
}
class Laboratorio {
    private String nombre;
    private int capacidad;
    private int nroMesas;
    private int nroSillas;
    public Laboratorio(String nombre, int capacidad, int nroMesas, int nroSillas) {
        this.nombre = nombre;
        this.capacidad = capacidad;
        this.nroMesas = nroMesas;
        this.nroSillas = nroSillas;
    }
    public void cantidadDeMuebles() {
        System.out.println("La cantidad de muebles es de: " + (nroMesas + nroSillas));
    }
    public void mostrar() {
        System.out.println("El laboratorio se llama: " + nombre +  ", la capacidad es de: " + capacidad + ", la cantidad de mesas es: " + nroMesas + ", y la cantidad de sillas es de: " + nroSillas);
    }
}